pkg/v1/remote/options.go

```diff
@@ -38,6 +38,7 @@ type options struct {
 	userAgent                      string
 	allowNondistributableArtifacts bool
 	updates                        chan<- v1.Update
+	pageSize                       int
```

Comment:
Set this to 1000 in `makeOptions` below, then we can drop the `else` clause in `list.go` and `catalog.go`.

If a user doesn't set page size, it defaults to 1000. 
If a user sets page size <= 0, we omit `n` altogether.
If a user sets page size > 0, we'll use that.